lib/index.js

```diff
@@ -23,11 +23,11 @@ exports.clone = function (obj, seen) {
         return obj;
     }
 
-    seen = seen || { orig: [], copy: [] };
+    seen = seen || new Map();
 
-    const lookup = seen.orig.indexOf(obj);
-    if (lookup !== -1) {
-        return seen.copy[lookup];
+    const lookup = seen.get(obj);
+    if (lookup) {
```

Comment:
Would this check be more appropriate to be `if (lookup === undefined) {`? According to the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get) when the value isn't present `undefined` is returned.

That said I'm not sure what edge case could cause an item to return a falsey value when it is found from the map in this case.